packages/type-extrinsics/src/SignableExtrinsic.ts

```diff
@@ -0,0 +1,28 @@
+// Copyright 2017-2018 @polkadot/extrinsics authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import { Extrinsic, Index, UncheckedExtrinsic } from '@polkadot/api-codec/index';
+import { KeyringPair } from '@polkadot/util-keyring/types';
+import u8aConcat from '@polkadot/util/u8a/concat';
+
+export default class SignableExtrinsic extends UncheckedExtrinsic {
+  sign (sender: KeyringPair, nonce: Index): Extrinsic {
+    // Prepend sender's publicKey and nonce
+    const message = u8aConcat(
+      sender.publicKey(),
```

Comment:
We need a `sender.address` that returns an Address.

Reply:
I was afraid of that. And got a snif of that in the UI today.Basically we probably need to move coded to the common repo then, which screws us up a bit. (since deps on repos are `dev` -> `common` -> `api` -> `apps/client`). As a stopgap,

```
new Address(sender.publicKey())
```

Not against doing the move, only at this point it is easier having it in here...